schematic/store/synapse_tracker.py

```diff
@@ -0,0 +1,103 @@
+"""This script is responsible for creating a 'pull through cache' class that can be
```

Comment:
This file is the main star of all these changes. Please read carefully.

Reply:
This question is probably due to how little time I ahve spent in this module. I'm curious why this was needed in the first place? Was Schematic repeatedly downloading the same entity in the same API call?

Reply:
> Was Schematic repeatedly downloading the same entity in the same API call?

Yes. Early in the script metadata about an entity was retrieved from Synapse. We can take the manifest FileEntity for example. Every time that manifest needed to be referenced the metadata from Synapse was re-retrieved.

In the benchmarking that I had done it cut down a significant number of calls to Synapse. In the screenshot I ran all tests prefixed with `test_submit_%` and the total number of HTTP calls went from 972 -> 594.
![image](https://github.com/user-attachments/assets/41da5fec-2b83-401e-91a9-cc410c08ba34)